packages/appium/lib/config.js

```diff
@@ -209,6 +217,42 @@ function getNonDefaultServerArgs (args) {
   return _.pickBy(args, (__, key) => isNotDefault(key));
 }
 
+/**
+ * Compacts an object for {@link showConfig}:
+ * 1. Removes `subcommand` key/value
+ * 2. Removes `undefined` values
+ * 3. Removes empty objects (but not `false` values)
+ * Does not operate recursively.
+ */
+const compactConfig = _.partial(
+  _.omitBy,
+  _,
+  (value, key) => key === 'subcommand' || _.isUndefined(value) || (!_.isBoolean(value) && _.isEmpty(value))
```

Comment:
isEmpty is also true for raw numbers

Reply:
cannot we simply reuse the lodash's compact method here?

Reply:
`_.isEmpty()` is true for booleans.  I'll fix it for numbers.  this probably needs a test

Reply:
ok, I've added a test here and fixed the issue with numeric values.  using `_.compact([value])` doesn't work, because `_.compact` does not recognize an empty object as something it should omit.